cli/cmd/lib_cluster_config.go

```diff
@@ -18,194 +18,156 @@ package cmd
-	// Next check AWS CLI config file
-	accessKeyID, secretAccessKey, err := aws.GetCredentialsFromCLIConfigFile()
-	if err == nil {
-		awsCreds.AWSAccessKeyID = accessKeyID
-		awsCreds.AWSSecretAccessKey = secretAccessKey
-		return nil
+	err = clusterconfig.InstallPrompt(clusterConfig, awsCreds.AWSAccessKeyID, awsCreds.AWSSecretAccessKey)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	// Next check Cortex CLI config file
-	cliConfig, errs := readCLIConfig()
-	if !errors.HasErrors(errs) && cliConfig != nil && cliConfig.AWSAccessKeyID != "" && cliConfig.AWSSecretAccessKey != "" {
-		awsCreds.AWSAccessKeyID = cliConfig.AWSAccessKeyID
-		awsCreds.AWSSecretAccessKey = cliConfig.AWSSecretAccessKey
-		return nil
+	if clusterConfig.Spot != nil && *clusterConfig.Spot {
+		clusterConfig.AutoFillSpot()
 	}
 
-	// Prompt
-	err = cr.ReadPrompt(awsCreds, awsCredentialsPromptValidation)
+	err = clusterConfig.Validate()
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
-	return nil
+	confirmClusterConfig(clusterConfig, awsCreds)
+
+	return clusterConfig, awsCreds, nil
 }
 
-// awsCreds is what was read from the cluster config YAML, after being passed through setInstallAWSCredentials() (so those credentials should be set)
-func setOperatorAWSCredentials(awsCreds *AWSCredentials) error {
-	// First check env vars
-	if os.Getenv("CORTEX_AWS_ACCESS_KEY_ID") != "" && os.Getenv("CORTEX_AWS_SECRET_ACCESS_KEY") != "" {
-		awsCreds.CortexAWSAccessKeyID = os.Getenv("CORTEX_AWS_ACCESS_KEY_ID")
-		awsCreds.CortexAWSSecretAccessKey = os.Getenv("CORTEX_AWS_SECRET_ACCESS_KEY")
-		return nil
-	}
-	if os.Getenv("CORTEX_AWS_ACCESS_KEY_ID") == "" && os.Getenv("CORTEX_AWS_SECRET_ACCESS_KEY") != "" {
-		return errors.New("Only $CORTEX_AWS_SECRET_ACCESS_KEY is set; please run `export CORTEX_AWS_ACCESS_KEY_ID=***`")
-	}
-	if os.Getenv("CORTEX_AWS_ACCESS_KEY_ID") != "" && os.Getenv("CORTEX_AWS_SECRET_ACCESS_KEY") == "" {
-		return errors.New("Only $CORTEX_AWS_ACCESS_KEY_ID is set; please run `export CORTEX_AWS_SECRET_ACCESS_KEY=***`")
+func getUpdateClusterConfig(awsCreds *AWSCredentials) (*clusterconfig.ClusterConfig, error) {
+	cachedClusterConfig := &clusterconfig.ClusterConfig{}
+	err := readClusterConfigFile(cachedClusterConfig, cachedClusterConfigPath)
+	if err != nil {
+		return nil, err
 	}
 
-	// Next check what was read from cluster config YAML
-	if awsCreds.CortexAWSAccessKeyID != "" && awsCreds.CortexAWSSecretAccessKey != "" {
-		return nil
-	}
-	if awsCreds.CortexAWSAccessKeyID == "" && awsCreds.CortexAWSSecretAccessKey != "" {
-		return errors.New(fmt.Sprintf("Only cortex_aws_secret_access_key is set in %s; please set cortex_aws_access_key_id as well", flagClusterConfig))
-	}
-	if awsCreds.CortexAWSAccessKeyID != "" && awsCreds.CortexAWSSecretAccessKey == "" {
-		return errors.New(fmt.Sprintf("Only cortex_aws_access_key_id is set in %s; please set cortex_aws_secret_access_key as well", flagClusterConfig))
+	userClusterConfig := &clusterconfig.ClusterConfig{}
```

Comment:
I think this can be `var userClusterConfig *clusterconfig.ClusterConfig` although I'm not 100% sure, what do you think?